### DISCUSSION TITLE
Is it expected that larger roi_size in Sliding Window Inferer typically improves outcome

### DISCUSSION THREAD

**Discussion by dyollb** [Project-MONAI/MONAI#7613] [Q&A] [answered]
I recently noticed some artifacts in my inference results. The center was pretty good, but outside a consistent width I was getting garbage.

![image](https://github.com/Project-MONAI/MONAI/assets/12702862/11f28cc3-8148-4b0a-b13e-7ec39172d79f)
outside the center region, parts of the brain are labeled as muscle or fat.

Because of this width issue, I guessed it might be related to the Sliding Window Inferer.
- I was using roi_size=[96, 96, 96] (same size as the patches I used for training)
- increasing the roi_size to [160, 160, 160] "fixed" the issue
- in fact, the validation mean dice improved for every single image in my dataset (evaluated on 163 subjects)
- the histogram shows the mean dice improvement for roi_size increasing 96 -> 128 -> 160

![mean_dice_compare_sliding_window_roi_size](https://github.com/Project-MONAI/MONAI/assets/12702862/83f105dc-8c99-4fc9-bcee-c52469035b12)

Is this a known behavior? This result suggests I should always use the larger roi_size.

**Comment by KumoLiu** [comment, marked as answer]
Hi @dyollb,

The `roi_size` parameter in the Sliding Window Inferer refers to the spatial window size for inferences. This size stands for the region of interest that the model will evaluate during the sliding window operation.
If `roi_size` is the same as your training patches (in your case, [96, 96, 96]), it means your inference operation happens exactly on the same spatial scales as during training.
Increasing the `roi_size` to a larger dimension (like [160, 160, 160]) expands the context that the model sees during the inference operation. This can often improve the model's performance because it is considering a larger region of the image and can therefore make better predictions, especially for peripheral regions of the image. This explains the improved validation mean dice for every single image in your dataset when `roi_size` is increased.
Therefore, selecting a suitable `roi_size` that balances computation efficiency and model performance is an important consideration in the Sliding Window Inference process. Typically, a larger `roi_size` provides more spatial context for better performance but also requires more computational resources. The histogram you provided revealing the mean dice improvement for increasing roi_size from 96 -> 128 -> 160 illustrates this performance enhancement with bigger `roi_size`.
You also noticed that outside the central region, parts of the brain were labeled as muscle or fat. This could be due to the model failing to correctly interpret these regions, likely due to insufficient spatial context during training or oversights in data labeling. Adjusting the `roi_size` can influence this, as it changes the region the model views for making predictions during inference, potentially rectifying incorrect labels in the periphery of the image.

Hope it helps,
Thanks

**Reply by dyollb** [reply]
Thanks for the explanation. Makes sense.
